apps/user-profile/src/app/pages/settings-page/settings-authorization/mfa-settings/mfa-settings.component.ts

```diff
@@ -0,0 +1,428 @@
+/* eslint-disable
+   @typescript-eslint/no-explicit-any,
+   @typescript-eslint/explicit-module-boundary-types,
+   @typescript-eslint/no-unsafe-member-access,
+   @typescript-eslint/no-unsafe-assignment */
+
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { now } from 'moment-timezone';
+import { AttributesManagerService } from '@perun-web-apps/perun/openapi';
+import { AuthService, MfaApiService, StoreService } from '@perun-web-apps/perun/services';
+import { OAuthService } from 'angular-oauth2-oidc';
+import { HttpClient } from '@angular/common/http';
+import { MatSlideToggle } from '@angular/material/slide-toggle';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'perun-web-apps-mfa-settings',
+  templateUrl: './mfa-settings.component.html',
+  styleUrls: ['./mfa-settings.component.scss'],
+})
+export class MfaSettingsComponent implements OnInit {
+  @ViewChild('toggle') toggle: MatSlideToggle;
+
+  mfaAvailable = false;
+  loadingMfa = false;
+
+  enforceMfa: boolean;
+  showDetail = false;
+  loadingCategories = false;
+  includeCategories: string[] = [];
+  excludeRps: string[] = [];
+  allCategories = false;
+  unchangedSettings = true;
+  unchangedEnforce = true;
+
+  categories = {};
+
+  constructor(
+    private attributesManagerService: AttributesManagerService,
+    private translate: TranslateService,
```

Comment:
Since it is used in html
```suggestion
    public translate: TranslateService,
```

Reply:
I agree that this is NOT the most beautiful component that I've ever seen :/. I tried to not duplicate the code and also not write infinity functions, but handling these nested toggles generated according to the "unstructured" object from API is not really good solution.